src/Symfony/Component/Workflow/Utils/PlaceEnumerationUtils.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Utils;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class PlaceEnumerationUtils
+{
+    public static function getPlaceKey(string|\UnitEnum $place): string
+    {
+        if ($place instanceof \UnitEnum) {
+            return \get_class($place).'::'.$place->name;
+        }
+
+        return $place;
+    }
+
+    public static function getTypedValue(string $place): string|\UnitEnum
+    {
+        try {
+            return \constant($place);
```

Comment:
should we look for `::` in the string first to avoid trying to load constants in such case ?

Reply:
Looking at this class, I really wonder if we shouldn't limit this feature to backed enums. If a developer does not want us to serialize an enum they declared, we should not guess how to serialize it anyway.

Reply:
At first I thought of only add support of BackedEnums and being able to use `BackedEnum::tryFrom` but in any case, I've been confronted to the problem that in the marking store, we don't really have any way to guess which enumeration we should use to `tryFrom` :confused:

Reply:
> which enumeration we should

How many enumerations should a user be allowed to use for a single workflow? Should they be allowed to mix and match values from different enums or even enum values and scalars?

Reply:
It's a choice I had a hard time making :sweat_smile: I'm torn between "we can do it, why limit ourselves" and "it doesn't make sense to allow 3 different enumerations and scalars for a workflow". So I proposed this PR by accepting the most permissive solution, but if you feel that this is not the right option, I will gladly make the changes :smile: